core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -2285,6 +2290,18 @@ private CompiledQuery repairTables(SqlExecutionContext executionContext) throws
         return compiledQuery.ofRepair();
     }
 
+    private void schedulePurgeO3Partitions(CharSequence tableName, int partitionBy) {
+        final MPSequence seq = messageBus.getO3PurgeDiscoveryPubSeq();
+        long cursor = seq.next();
+        if (cursor > -1) {
```

Comment:
because queue is shared it would be good to look out for -2  (CAS failure) and retry rather than erroring out

Reply:
Changed in all 3 places where similar async job scheduling is done